### PR TITLE
[CMLG- 021] Return appropriate result

### PR DESCRIPTION
**Pull request by ywu666**
Issue:
Try to return the appropriate data based on the display setting that the user chooses(Display all the data at once or display specific number of rows per page).  

Solution:
Add pageNum parameter to the API. If the pageNum is null, then display all the data at once. Otherwise, display the data of certain pageNum. 

Risk: 
Need some official test cases to test the result. 
The pageNum parameter should only be an integer which is between 1 and the total page number.
The pageRows parameter should either be an integer or a word "all".

Reviewed By: 
Emily, Kevin, Annie, Eileen, Dave

### CODE REVIEW

File: TranslationBackend/app/Http/Controllers/WordController.php [hunk 1]
```diff
@@ -44,15 +44,31 @@ public function show()
...
+
+            // default page number is 1
+            $pageNum = 1;
+            if(request() -> has( 'pageNum')) {
+                $pageNum = request()->pageNum;
+            }
+
+            // default rows in a page is 10
+            $pageRows = 10;
+            if(request() -> has( 'pageRows')) {
```
Comment (annithinggoes) [inline TranslationBackend/app/Http/Controllers/WordController.php:60]: Yea idk how i feel about the pageRows being in the endpoint, it doesn't seem right to put it there.....

File: TranslationBackend/app/Word.php [hunk 2]
```diff
@@ -8,18 +8,22 @@
...
-        if ($word == null) {
-            $translation = $translation->take(10); 
-        }        
+
+        if($pageRows == 'all' ){
+            $translation = $allTranslations;
+        }else{
+            // the first take gets the data from 1 up to $pageNum * $pageRows
+            // the second data gets the last $pageRows data
+            $translation = $allTranslations -> take($pageNum * $pageRows) -> take(-$pageRows);
```
Comment (annithinggoes) [inline TranslationBackend/app/Word.php:25]: curious why you didn't use skip()

File: TranslationBackend/resources/.env.example [hunk 3]
```diff
@@ -0,0 +1,46 @@
+APP_NAME=Laravel
```
Comment (annithinggoes) [inline TranslationBackend/resources/.env.example:1]: Why is this being pushed up????? Is this not on master?

File: TranslationBackend/app/Word.php [hunk 4]
```diff
@@ -8,18 +8,22 @@
...
             ->where('words.name', 'like', '%'.$word.'%')
             ->distinct()
             ->get();
-       
-        if ($word == null) {
-            $translation = $translation->take(10); 
-        }        
+
+        if($pageRows == 'all' ){
+            $translation = $allTranslations;
```
Comment (annithinggoes) [inline TranslationBackend/app/Word.php:21]: Are we lazy loading this? Or is this future stuff

**Review by annithinggoes — COMMENTED** [review]
Have a look at my comments.

**Comment by zyan225** [comment]
> Have a look at my comments. Also did you not update the api.php to include the new updated endpoint url?

Looks like endpoint in api.php file doesn't need to specify parameters?

**Comment by annithinggoes** [comment]
> > Have a look at my comments. Also did you not update the api.php to include the new updated endpoint url?
> 
> Looks like endpoint in api.php file doesn't need to specify parameters?

Oh wait nvm yea gotcha

File: TranslationBackend/app/Word.php (hunk 2 above)
Comment (jge385) [reply TranslationBackend/app/Word.php:25]: lol haven't heard this function, just googled it... Thx~ will change this tomorrow

File: TranslationBackend/app/Word.php (hunk 4 above)
Comment (ywu666) [reply TranslationBackend/app/Word.php:21]: No, just simply return all data.

**Review by annithinggoes — APPROVED** [review]

**Review by EileenSiqi — APPROVED** [review]

**Comment by ywu666** [comment]
Issue:
Try to return the appropriate data based on the display setting that the user chooses(Display all the data at once or display specific number of rows per page).  

Solution:
Add pageNum parameter to the API. If the pageNum is null, then display all the data at once. Otherwise, display the data of certain pageNum. 

Risk: 
Need some official test cases to test the result. 
The pageNum parameter should only be an integer which is between 1 and the total page number.
The pageRows parameter should either be an integer or a word "all".

Reviewed By: 
Emily, Kevin, Annie, Eileen, Dave

**Pull request merged**